src/cli.ts

```diff
@@ -1,51 +1,537 @@
 #!/usr/bin/env node
 import 'source-map-support/register';
 
-import * as dns from 'dns';
-
-import * as commander from 'commander';
 import * as log from 'loglevel';
+import * as yargs from 'yargs';
 
-import { isArgFormatInvalid } from './helpers/helpers';
+import {
+  checkInternet,
+  getProcessEnvs,
+  isArgFormatInvalid,
+} from './helpers/helpers';
 import { supportedArchs, supportedPlatforms } from './infer/inferOs';
 import { buildNativefierApp } from './main';
 import { NativefierOptions } from './options/model';
-import { parseBooleanOrString, parseJson } from './utils/parseUtils';
+import { parseJson } from './utils/parseUtils';
+import { DEFAULT_ELECTRON_VERSION } from './constants';
+
+export function initArgs(argv: string[]): yargs.Argv<any> {
+  const args = yargs(argv)
+    .scriptName('nativefier')
+    .usage(
+      '$0 <targetUrl> [outputDirectory] [other options]\nor\n$0 --upgrade <pathToExistingApp> [other options]',
+    )
+    .example(
+      '$0 <targetUrl> -n <name>',
+      'Make an app from <targetUrl> and set the application name to <name>',
+    )
+    .example(
+      '$0 --upgrade <pathToExistingApp>',
+      'Upgrade (in place) the existing Nativefier app at <pathToExistingApp>',
+    )
+    .example(
+      '$0 <targetUrl> -p <platform> -a <arch>',
+      'Make an app from <targetUrl> for the os <platform> and cpu architecture <arch>',
+    )
+    .positional('targetUrl', {
+      describe:
+        'The URL that you wish to to turn into a native app. Required if not using --upgrade',
+      type: 'string',
+    })
+    .positional('outputDirectory', {
+      defaultDescription: 'current directory',
+      describe: 'The directory to output the generated app to',
+      normalize: true,
+      type: 'string',
+    })
+    // App Creation Options
+    .option('a', {
+      alias: 'arch',
+      choices: supportedArchs,
+      defaultDescription: "current node's compiled cpu architecture",
+      description: 'The cpu architecture to build for',
+      group: 'App Creation Options',
```

Comment:
Two things:

1. Let's extract groups into a few consts, to avoid a typo in a group name accidentally creating a spurious extra group
2. What about adding a few visible extra "delimiter-ish" chars at the end of each group, to improve readability by making groups super visible? E.g. `group: 'App Creation Options ====='. Should not be too long, else we'll get into wrapping issues.
    1. Or (if offered by `yargs`) can groups be bolded/colored by `yargs` ?

Reply:
I think I have an easier solution for 1. Instead of specifying them in the option you can declare a group of options like:
```javascript
.group(['flash', 'flash-path'], 'Flash Options (DEPRECATED)')
```

And then future options would just need to be added to their respective group's array.

For 2, it looks like yargs supports Chalk command line coloring (if we want to `npm install chalk`, see https://github.com/yargs/yargs/issues/251), as well it looks like we could monkey with the category names to achieve what we want like so:

```
*****************
* Debug Options *
*****************
      --crash-reporter  remote server URL to send crash reports                           [string]
      --verbose         enable verbose/debug/troubleshooting logs       [boolean] [default: false]
```

Reply:
<Usual blah about limiting adding deps>, `chalk` is a total tree of 6 deps. What do you think of something dumber like my original suggestion, of using a few pre/post characters (like `=====` or `-----` or `*****`) to do the job?

Reply:
I'm mostly fine with it. To be clear, my example above is not using chalk, just pre/post chars.

Reply:
> I'm mostly fine with it. To be clear, my example above is not using chalk, just pre/post chars.

@TheCleric ooooh okay, alright then. To keep on picking nits, I see the intent of

```
*****************
* Debug Options *
*****************
```

, but it goes a bit far in the other direction, causing 3 lines per section instead of 1. I'm partial to either 

```
Debug Options =====
```

or

```
Debug Options
=============
```

Reply:
> ```
> ===== Debug Options =====
> ```

I like this one the best! 😄